readiness-tool/popup.js

```diff
@@ -0,0 +1,238 @@
+  supportedAnalytics = document.getElementById('analytics-supported');
+  notSupportedAds = document.getElementById('ads-notSupported');
+  notSupportedAnalytics = document.getElementById('analytics-notSupported');
+  supportedAds.innerHTML = supportedAnalytics.innerHTML =
+    notSupportedAds.innerHTML = notSupportedAnalytics.innerHTML = 
+    loadingMessage;
+  // Gets the DOM of the current web page and converts it to a string
+  chrome.tabs.executeScript(null, {
+    file: 'getPagesSource.js',
+  }, function () {});
+};
+/**
+ * Returns all the 3rd party applications found on the website
+ * @param {String} html - String containing all HTML on the page
+ * @return {Object} 
+ */
+function findDetectedApps(html) {
+  fetch('apps.json')
+    .then(function (response) {
+    if (response.status >= 400) {
+      console.error('Fetching apps.json failed with this status Code: ' + 
+                  response.status);
+      return;
+    }
+    if (response.statusIsOK) {
+        console.error(response.status);
+      }
+    if (!response.statusIsOK) {
+      // Examine the text in the response  
+      response.json().then(function (data) {
+        let listAllApps = data.apps;
+        detectedApps = filteredApps(html, listAllApps);
+        showSupportedAppsInView(detectedApps, listAllApps);
+        return detectedApps;
+      });
+    }
+    
+  }).catch(function (err) {
+    console.log('Fetch Error :-S', err);
+  });
+}
+/**
+ * Add supported and unsupported applications to the view
+ * @param {Objecte} detectedApps - All 3rd Party Applications found on page
+ * @param {!Object} listAllApps - JSON of all 3p vendors
+ */
+function showSupportedAppsInView(detectedApps, listAllApps) {
+  supportedAds.innerHTML = supportedAnalytics.innerHTML =
+    notSupportedAds.innerHTML = notSupportedAnalytics.innerHTML = blankMessage;
+  supportedAds.appendChild(
+    makeList(detectedApps.supported.ads, false, listAllApps));
+  supportedAnalytics.appendChild(
+    makeList(detectedApps.supported.analytics, false, listAllApps));
+  notSupportedAds.appendChild(
+    makeList(detectedApps.notSupported.ads, true, listAllApps));
+  notSupportedAnalytics.appendChild(
+    makeList(detectedApps.notSupported.analytics, true, listAllApps));
+}
+/**
+ * Splits all detected apps into 'supported' and 'not supported'
+ * @param {String} htmlString - String containing all HTML on the page
+ * @param {!Object} listAllApps - JSON of all the 3p vendors 
+ * @return {Object} 
+ */
+function filteredApps(htmlString, listAllApps) {
+  const foundThis = {
+    'supported': {
+      'ads': [],
+      'analytics': []
+    },
+    'notSupported': {
+      'ads': [],
+      'analytics': []
+    }
+  };
+  // for all the app objects in the apps.JSON file
+  Object.keys(listAllApps).forEach(function (key) {
+    let val = listAllApps[key];
+    // If object has a 'script' key
+    if (val.script != null) {
+      Object.keys(val.script).forEach(function (x) {
```

Comment:
Looks like `val.script` is an array, not an object, right? 

You don't need to do `Object.keys()` for arrays (the keys are just indices), just do `val.script.forEach(...)`.